aquarium/aquarium.js

```diff
@@ -1701,6 +1704,94 @@ function initialize() {
+      elapsedTime = 0.0;
+    } else {
+      elapsedTime = now - then;
+    }
+    then = now;
+
+    if (g.net.sync) {
+      clock = now * g.globals.speed;
+      eyeClock = now * g.globals.eyeSpeed;
+    } else {
+      // we have our own clock.
+      clock += elapsedTime * g.globals.speed;
+      eyeClock += elapsedTime * g.globals.eyeSpeed;
+    }
+
+    frameCount++;
+    g_fpsTimer.update(elapsedTime);
+    fpsElem.innerHTML = g_fpsTimer.averageFPS;
+
+    if (g_shadersNeedUpdate) {
+      setShaders(true);
+      g_shadersNeedUpdate = false;
+    }
+
+    if (g_vrUi) {
+      // Set fps and prepare rendering it.
+      g_vrUi.setFps(g_fpsTimer.averageFPS);
+
+      // Query gamepad button clicked event.
+      g_vrUi.queryGamepadStatus();
+
+      // TODO: Support VRUI when doing multiview rendering.
+      if (!useMultiviewForStereo() && g_vrUi.isMenuMode) {
+
+        // When VR UI in menu mode, UI need a cursor to help user do select operation. Currently, cursor uses
+        // head-neck model which means a point in front of user and user could move the point by rotating their head(with HMD).
+        // A click event will be triggered when user stare at a label 2 seconds.
+        // TODO : add gamepad support to control cursor and trigger select event with VR controllers.
+
+        // Jquery selector description.
+        var selectorDescription;
+
+        // VR UI return whether there is an option been selected in VR mode.
+        var clickedLabel = g_vrUi.queryClickedLabel([0, 0, 0], g_frameData.pose.orientation);
+        if (clickedLabel != null) {
+          if (clickedLabel.isAdvancedSettings) {
+            selectorDescription = "#optionsContainer > div:contains(" + clickedLabel.name + ")";
+            $(selectorDescription).click();
+          } else if (clickedLabel.name == "options") {
+            $("#options").click();
+          } else {
+            selectorDescription = "#setSetting" + clickedLabel.name;
+            $(selectorDescription).click();
+          }
+        }
+      }
+    }
+
+    let glLayer = session.renderState.baseLayer;
+
+    gl.bindFramebuffer(gl.FRAMEBUFFER, glLayer.framebuffer);
+    gl.enable(gl.SCISSOR_TEST);
+
+    for (const view of pose.views) {
+      let xrViewport = glLayer.getViewport(view);
+      gl.viewport(xrViewport.x, xrViewport.y, xrViewport.width, xrViewport.height);
+      gl.scissor(xrViewport.x, xrViewport.y, xrViewport.width, xrViewport.height);
+      render(view.projectionMatrix, view.transform.matrix, false, view); // Not clear on what the final arg is used for here? Glancing at the code it looks like it should probably be view instead of pose.
+    }
+    //renderStereo(pose.views[0].projectionMatrix, pose.views[1].projectionMatrix, viewInverseTemp, vrPose);
+    // gl.clearColor(Math.cos(now / 2000),
+    //               Math.cos(now / 4000),
+    //               Math.cos(now / 6000), 1.0);
+
+    // gl.clear(gl.COLOR_BUFFER_BIT | gl.DEPTH_BUFFER_BIT);
```

Comment:
Let's removed these commented out calls unless they're trying to demonstrate some alternate technique. If so, they should be commented to indicate what they are trying to show.

Reply:
Also fixed